src/command_modules/azure-cli-hdinsight/azure/cli/command_modules/hdinsight/custom.py

```diff
@@ -173,23 +198,52 @@ def create_cluster(cmd, client, cluster_name, resource_group_name, cluster_type,
         # Specify storage account details only when storage arguments are provided
         storage_accounts.append(
             StorageAccount(
-                name=storage_account,
+                name=storage_account_endpoint,
                 key=storage_account_key,
                 container=storage_default_container,
                 file_system=storage_default_filesystem,
                 is_default=True
-            ))
+            )
+        )
+
+    additional_storage_accounts = []  # TODO: Add support for additional storage accounts
     if additional_storage_accounts:
         storage_accounts += [
             StorageAccount(
-                name=s.storage_account,
+                name=s.storage_account_endpoint,
                 key=s.storage_account_key,
                 container=s.container,
                 is_default=False
             )
             for s in additional_storage_accounts
         ]
 
+    cluster_identity = build_identities_info(assign_identity)
+    msi_resource_id = assign_identity and next((x for x in assign_identity if x != MSI_LOCAL_ID), None)
```

Comment:
Good Catch! Change the code to take the first element in `assign_identity`.

Reply:
In the future(this will happen very soon), we will accept 0~2 assign identities so it should be an array. But in this change, array is not necessary. If you feel this is not good, I can change it:)